execution/executor/src/mock_vm/mod.rs

```diff
@@ -70,65 +70,68 @@ impl VMExecutor for MockVM {
         let mut outputs = vec![];
 
         for txn in transactions {
-            match decode_transaction(&txn) {
-                Transaction::Mint { sender, amount } => {
-                    let old_balance = read_balance(&output_cache, state_view, sender);
-                    let new_balance = old_balance + amount;
-                    let old_seqnum = read_seqnum(&output_cache, state_view, sender);
-                    let new_seqnum = old_seqnum + 1;
-
-                    output_cache.insert(balance_ap(sender), new_balance);
-                    output_cache.insert(seqnum_ap(sender), new_seqnum);
-
-                    let write_set = gen_mint_writeset(sender, new_balance, new_seqnum);
-                    let events = gen_events(sender);
-                    outputs.push(TransactionOutput::new(
-                        write_set,
-                        events,
-                        0,
-                        KEEP_STATUS.clone(),
-                    ));
-                }
-                Transaction::Payment {
-                    sender,
-                    recipient,
-                    amount,
-                } => {
-                    let sender_old_balance = read_balance(&output_cache, state_view, sender);
-                    let recipient_old_balance = read_balance(&output_cache, state_view, recipient);
-                    if sender_old_balance < amount {
+            if let ExecutorTransaction::UserTransaction(user_txn) = txn {
```

Comment:
just do the `try_from` thing and unwrap?

Reply:
My plan is I'll get rid of the `expect` soon.